commit 1734291a7f73d7cb4edd9f0e00291def1de4498b (refs/changes/39/3539/3)
Author: Tim Kordas <tim.kordas@joyent.com>
Date:   2018-03-05T13:17:44-08:00 (1 year, 7 months ago)
    
    MANTA-3594 mahi crashed when processing 'admin' group membership after MANTA-3508

diff --git a/package.json b/package.json
index 8356b9b..5e5540f 100644
--- a/package.json
+++ b/package.json
@@ -31,7 +31,7 @@
     },
     "devDependencies": {
         "fakeredis": "0.1.3",
-        "nodeunit": "0.8.3",
+        "nodeunit": "0.9.1",
         "nodeunit-plus": "0.0.1"
     },
     "sdcDependencies": {
diff --git a/v1/lib/auth_cache.js b/v1/lib/auth_cache.js
index 7231340..8cb26e0 100644
--- a/v1/lib/auth_cache.js
+++ b/v1/lib/auth_cache.js
@@ -735,6 +735,15 @@ function removeGroupMember(self, userdn, groupdn, cb) {
                 user: payload
             }, 'got user entry');
             payload = JSON.parse(payload);
+            if (!payload.groups) {
+                // don't bother with client.set() we didn't do anything
+                log.info({
+                    key: key,
+                    entry: payload
+                }, 'user has no groups');
+                cb(err);
+                return;
+            }
             delete payload.groups[group];
             payload = JSON.stringify(payload);
             log.info({
